index.rst

```diff
@@ -1,61 +1,338 @@
+
+IDM-0205
+    When displaying the list of user-generated tokens, the date and time at which a user-generated token was last used must be shown alongside the token name.
+
+IDM-0206
+    Users must be notified via email of any change to their linked federated identities or any creation or deletion of a new user-generated token.
+
+Account management
+==================
+
+Status
+------
+
+IDM-1000
+    Accounts that are pending approval (under IDM-0003) can authenticate and see their account status and metadata page, but not access any other part of the Science Platform.
+
+IDM-1001
+    Administrators of the Science Platform must be able to freeze accounts.
+    Frozen accounts may be placed in a state where they cannot authenticate at all, or in a state where they can only see their account status and metadata page but no other part of the Science Platform.
+    A reason viewable by other administrators should be associated with a frozen account.
+    Frozen accounts still hold the account name and do not allow it to be reused.
+
+IDM-1002
+    Administrators of the Science Platform must be able delete accounts.
+    This is normally used for mistakenly-created accounts, not for accounts that were legitimate and active but should no longer be allowed access.
+
+IDM-1003
+    It must be possible to set an expiration date on an account.
+    This can be done by Science Platform administrators, or by the person approving access in the IDM-0003 use case.
+    When the expiration date arrives, the account must be automatically frozen.
+
+IDM-1004
+    Users must be notified via email of upcoming account expiration so that they can investigate renewal options if needed.
+
+Metadata
+--------
+
+IDM-1100
+    A full name must be associated with each account and prepopulated with information from the identity provider.
+    The user must be able to change the full name to anything they wish.
+
+IDM-1101
+    An email address must be associated with each account and prepopulated with information from the identity provider.
+    The user must be able to change the email address to anything they wish, but they must then verify that the email address is valid and owned by them by responding to a challenge sent to that email address.
+    Challenges for an email address must not contain user-provided content so that they cannot be used for spamming purposes.
+
+Quotas
+------
+
+IDM-1200
+    Users may have one or more quota grants associated directly with their account.
+    These may represent file storage quotas or any other service limit that may vary by user (API rate limits, CPU equivalents for batch jobs, download size limits, or whatever may eventually be appropriate).
+    The identity management system need not understand the quotas, but it should be able to sum multiple quotas under the same label.
```

Comment:
Is summing the quotas the right thing to do?  This seems a little out of place.  I can see wanting to see all your quotas and the status of all of them, but I think there would be different rules for each quota (like if it's a collaboration, you shouldn't just fill it with your personal stuff).

Reply:
The requirement just says that we have to have the capability. (Note that collaboration quotas and personal quotas are separate, as noted under groups, so they would apply to different paths.) I do think that summing quotas will be the correct action most of the time, but possibly not all of the time.